espresso/core/java/androidx/test/espresso/matcher/ViewMatchers.java

```diff
@@ -563,21 +563,44 @@ public int getValue() {
   public static <T> void assertThat(String message, T actual, Matcher<T> matcher) {
     if (!matcher.matches(actual)) {
       Description description = new StringDescription();
+      String mismatchDescriptionString = getMismatchDescriptionString(actual, matcher);
       description
           .appendText(message)
           .appendText("\nExpected: ")
           .appendDescriptionOf(matcher)
-          .appendText("\n     Got: ");
+          .appendText("\n     Got: ")
+          .appendText(mismatchDescriptionString);
       if (actual instanceof View) {
-        description.appendValue(HumanReadables.describe((View) actual));
-      } else {
-        description.appendValue(actual);
+        // Get the human readable view description.
+        String humanReadableViewDescription = HumanReadables.describe((View) actual);
+        // If the human readable view description isn't the same as the mismatch description,
+        // add it to the description.
+        if (!mismatchDescriptionString.equals(humanReadableViewDescription)) {
+          if (!mismatchDescriptionString.isEmpty()) {
+            // Mismatch description wasn't empty, this means that the current line in the
+            // description is:
+            //      Got: ${mismatchDescriptionString}
+            // We'll start a new line for the human readable line so the two don't get jumbled
+            // together.
+            description.appendText("\n          ");
+          }
+          description.appendText(humanReadableViewDescription);
+        }
       }
       description.appendText("\n");
       throw new AssertionFailedError(description.toString());
     }
   }
 
+  private static <T> String getMismatchDescriptionString(T actual, Matcher<T> matcher) {
+    Description mismatchDescription = new StringDescription();
+    matcher.describeMismatch(actual, mismatchDescription);
+    String mismatchDescriptionString = mismatchDescription.toString();
+    mismatchDescriptionString =
+        mismatchDescriptionString == null ? "" : mismatchDescriptionString.trim();
```

Comment:
`mismatchDescriptionString` will never be null since you're using `StringDescription`.

Reply:
(`describeMismatch` doesn't return null, at worst it doesn't call a method on description)